### PR TITLE
feat: add Aider-based linting for edit_file (agentskills)

### PR DESCRIPTION
**Pull request by tobitege**
### Done
- added only required files to runtime in new "aider" folder (incl. readme and license files)
- added unit tests for the Linter class itself; original code required several fixes to satisfy linter

### Open
WIP thus not checked in yet is the actual agentskills.py changes for edit_file to use Linter.

@neubig @rbren would love to get some direction of **expectations** for the new functionality, thanks!

### CODE REVIEW

File: opendevin/runtime/aider/README.md
Comment (neubig) [inline opendevin/runtime/aider/README.md:1]: We probably don't need all of this, could you reduce this to a link to the original Aider, and explain which parts we have adopted?

**Review by neubig — COMMENTED** [review]
Hey @tobitege , this looks great! In terms of expectations, this seems pretty good to me -- basically adding linting checks when we edit files.

One thing I considered is whether we should have this configurable in some way, but honestly I think that probably linting every time we make a file change is not a bad idea (as long as it's not too slow). I'd be happy to take a full look and test it out once `edit_file` is implemented too.

File: opendevin/runtime/aider/README.md
Comment (tobitege) [reply opendevin/runtime/aider/README.md:1]: Due to it being under a different license, I had put their files in a separate folder, incl. their license file.
How would you like me to proceed with that file structure layout in general with that regard?

**Comment by tobitege** [comment]
Closed in favor of https://github.com/OpenDevin/OpenDevin/pull/2489

**Pull request closed without merging**